import_tracker/rest.py

```diff
@@ -76,13 +81,18 @@ def getImports(query=None, user=None, unique=False, limit=None, offset=None, sor
     return processCursor(cursor, user)
 
 
-def moveLeafFiles(folder, user, assetstore):
+def moveLeafFiles(folder, user, assetstore, progress, job):
+    if job['status'] == JobStatus.CANCELED:
+        raise ImportTrackerCancelError()
+
     Folder().updateFolder(folder)
 
+    unique_clause = {'assetstoreId': {'$ne': ObjectId(assetstore['_id'])}}
     folder_item = Item().findOne({
         'folderId': folder['_id'],
     })
-    unique_clause = {'assetstoreId': {'$ne': ObjectId(assetstore['_id'])}}
+    if not folder_item:
+        raise RestException('Folder %s has no item' % folder['_id'])
```

Comment:
Folders don't have to have items, and it shouldn't stop processing other folders.